src/Temporalio/Worker/TemporalWorkerOptions.cs

```diff
@@ -109,15 +109,26 @@ public TemporalWorkerOptions()
         /// If unset, the default is
         /// <c>Assembly.GetEntryAssembly().ManifestModule.ModuleVersionId</c>.
         /// </summary>
+        /// <remarks>Exclusive with <see cref="DeploymentOptions"/>.</remarks>
+        [Obsolete("Use DeploymentOptions instead")]
         public string? BuildId { get; set; }
 
         /// <summary>
         /// Gets or sets a value indicating whether this worker opts into the worker versioning feature. This ensures it
         /// only receives workflow tasks for workflows which it claims to be compatible with. The
         /// <see cref="BuildId"/> field is used as this worker's version when enabled, and must be set.
         /// </summary>
+        /// <remarks>Exclusive with <see cref="DeploymentOptions"/>.</remarks>
+        [Obsolete("Use DeploymentOptions instead")]
         public bool UseWorkerVersioning { get; set; }
 
+        /// <summary>
+        /// Gets or sets the deployment options for this worker.
+        /// </summary>
+        /// <remarks>WARNING: Deployment-based versioning is experimental and APIs may
+        /// change.</remarks>
+        public WorkerDeploymentOptions? DeploymentOptions { get; set; }
```

Comment:
```suggestion
        public WorkerDeploymentOptions? Deployment { get; set; }
```

In cases where we have sub-options, we don't usually put the `Options` suffix on the property name (but maybe we do inconsistently somewhere I didn't check)

Reply:
I think it makes more sense here, because of the use versioning flag inside. Deployment implies just the name part